src/ConfigWrapper/WindowsRegistryConfigWrapper.cs

```diff
@@ -185,16 +222,26 @@ public void Set<T>(string key, T value, bool createKeyIfNeeded)
         /// <inheritdocs />
         public void Set<T>(string key, T value)
         {
-            using (var regKey = this.GetParentKey(key, false, true))
-            {
-                regKey.SetValue(this.GetKeyAsArray(key).Last(), value.ToString());
-                regKey.Close();
-            }
+            this.Set<T>(key, value, false);
```

Comment:
Maybe to add another Set<T> method with the parameter isFullyQualified. To use the FullKey or just the key

Reply:
Yeah that's interesting.